src/test/ui/associated-types/defaults-cyclic-fail.rs

```diff
@@ -0,0 +1,37 @@
+// compile-fail
+
+#![feature(associated_type_defaults)]
+
+// Having a cycle in assoc. type defaults is okay...
+trait Tr {
+    type A = Self::B;
+    type B = Self::A;
+}
+
+// ...but is an error in any impl that doesn't override at least one of the defaults
+impl Tr for () {}
+//~^ ERROR overflow evaluating the requirement
+
+// As soon as at least one is redefined, it works:
+impl Tr for u8 {
+    type A = u8;
```

Comment:
is there a cpass test for this? I think an actual `defaults-cyclic-pass` test would be a good idea.